x-pack/plugins/security/server/lib/authorization/register_privileges_with_cluster.js

```diff
@@ -7,16 +7,12 @@
 import { difference, isEmpty, isEqual } from 'lodash';
 import { buildPrivilegeMap } from './privileges';
 import { getClient } from '../../../../../server/lib/get_client_shield';
-import { actionsFactory } from './actions';
-
-
 
 export async function registerPrivilegesWithCluster(server) {
-  const config = server.config();
 
-  const actions = actionsFactory(config);
-  const application = config.get('xpack.security.rbac.application');
-  const savedObjectTypes = server.savedObjects.types;
+  const { authorization } = server.plugins.security;
+  const { types: savedObjectTypes } = server.savedObjects;
+  const { actions, application } = authorization;
```

Comment:
nit: we could simplify the destructuring here by passing `plugin` as a second argument, then this just becomes `const { actions, application } = plugin.authorization`. Feel free to ignore me here.

Reply:
I'm gonna leave this alone for now... I really dislike having this dependent on the whole `server`, and would prefer that we pass in the explicit dependencies, and we can address it at this point.